backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ItemStockSammMapper.java

```diff
@@ -0,0 +1,277 @@
+                materialItemStock.getCustomerOrderId(), materialItemStock.getCustomerOrderPositionId());
+            position.setOrderPositionReference(opr);
+        }
+        ItemQuantityEntity itemQuantityEntity = new ItemQuantityEntity(materialItemStock.getQuantity(),
+            materialItemStock.getMeasurementUnit());
+        AllocatedStock allocatedStock = new AllocatedStock(itemQuantityEntity, materialItemStock.getLocationBpns(),
+            materialItemStock.isBlocked(), materialItemStock.getLocationBpna());
+        var allocatedStocksList = new ArrayList<AllocatedStock>();
+        allocatedStocksList.add(allocatedStock);
+        position.setAllocatedStocks(allocatedStocksList);
+        return samm;
+    }
+
+    public ItemStockSAMM toItemStockSamm(ProductItemStock productItemStock) {
+        ItemStockSAMM samm = new ItemStockSAMM();
+        samm.setDirection(DirectionCharacteristic.OUTBOUND);
+        samm.setPositions(new ArrayList<>());
+        samm.setMaterialGlobalAssetId(productItemStock.getMaterial().getMaterialNumberCx());
+        samm.setMaterialNumberSupplier(productItemStock.getMaterial().getOwnMaterialNumber());
+        samm.setMaterialNumberCustomer(mprService.find(productItemStock.getMaterial(),
+            productItemStock.getPartner()).getPartnerMaterialNumber());
+        var posList = new ArrayList<Position>();
+        samm.setPositions(posList);
+        Position position = new Position();
+        if (productItemStock.getCustomerOrderId() != null || productItemStock.getCustomerOrderPositionId() != null
+            || productItemStock.getSupplierOrderId() != null) {
+            OrderPositionReference opr = new OrderPositionReference(productItemStock.getSupplierOrderId(),
+                productItemStock.getCustomerOrderId(), productItemStock.getCustomerOrderPositionId());
+            position.setOrderPositionReference(opr);
+        }
+        ItemQuantityEntity itemQuantityEntity = new ItemQuantityEntity(productItemStock.getQuantity(),
+            productItemStock.getMeasurementUnit());
+        AllocatedStock allocatedStock = new AllocatedStock(itemQuantityEntity, productItemStock.getLocationBpns(),
+            productItemStock.isBlocked(), productItemStock.getLocationBpna());
+        var allocatedStocksList = new ArrayList<AllocatedStock>();
+        allocatedStocksList.add(allocatedStock);
+        position.setAllocatedStocks(allocatedStocksList);
+        return samm;
+    }
+
+    public List<ReportedProductItemStock> sammToReportedProductItemStock(ItemStockSAMM samm, Partner partner) {
+        String matNbrCustomer = samm.getMaterialNumberCustomer();
+        String matNbrSupplier = samm.getMaterialNumberSupplier(); // should be ownMaterialNumber
+        String matNbrCatenaX = samm.getMaterialGlobalAssetId();
+        ArrayList<ReportedProductItemStock> outputList = new ArrayList<>();
+        if(samm.getDirection() != DirectionCharacteristic.INBOUND) {
```

Comment:
The Supplier sends his product stocks to us. From his point of view this is OUTBOUND stock. For the direction the sending party's view is the leading one :)

Reply:
Hmm...,, no I think it's correct the way it is. 

We are unmarshalling a ReportedProductItemStock here. That means, it came from a message, that a customer sent to his supplier, telling the supplier how many materials are in storage at the customers side at this moment. 

That means, from the senders side (the customer), it was a MaterialItemStock, that was marshalled into a SAMM. Therefore the customer will have to set an INBOUND flag when sending his message. Also see the toItemStock method which takes a MaterialItemStock as input. (Line 49). 

Do you agree?

Reply:
Got it, got confused by inversion between my material vs partner's product. The Test Case was addressing the ReportedMaterialItemStock mapping not the ReportedProdcutItemStock.

The ReportedProductItemStock represents the old PartnerProductStock. Thus, Direction Inbound is correct.